04-data-structures-part1.Rmd

```diff
@@ -2,336 +2,271 @@
 layout: page
 title: R for reproducible scientific analysis
 subtitle: Data structures
-minutes: 45
+minutes: 90
 ---
 
 ```{r, include=FALSE}
 source("tools/chunk-options.R")
 ```
 
-
 > ## Learning Objectives {.objectives}
 >
 > - To be aware of the different types of data
-> - To be aware of the different basic data structures commonly encountered in R
+> - To begin exploring the data.frame, and understand how it's related to vectors, factors and lists
 > - To be able to ask questions from R about the type, class, and structure of an object.
 >
 
-## Data Types
+One of R's most powerful features is its ability to deal with tabular data - like what you might already have in a spreadsheet or a CSV. Let's start by making a toy dataset in your `data/` directory, called `feline-data.csv`:
 
-Before we can analyse any data, we'll need to have a strong
-understanding of the basic data types and data structures. It is **Very
-Important** to understand because these are the things you will
-manipulate on a day-to-day basis in R, and are the source of most
-frustration encountered by beginners.
+```{r, eval=FALSE}
+coat,weight,likes_string
+calico,2.1,TRUE   
+black,5.0,FALSE   
+tabby,3.2,TRUE 
+```
 
-R has 5 basic atomic types (meaning they can't be broken down into anything smaller):
+We can load this into R via the following:
 
-* logical (e.g., `TRUE`, `FALSE`)
-* numeric
-    * integer (e.g, `2L`, `as.integer(3)`)
-    * double (i.e. decimal) (e.g, `-24.57`, `2.0`, `pi`)
-* complex (i.e. complex numbers) (e.g, `1 + 0i`, `1 + 4i`)
-* text (called "character" in R) (e.g, `"a"`, `"swc"`, `'This is a cat'`)
+```{r}
+cats <- read.table(file="data/feline-data.csv", header=TRUE, sep=",")
+cats
+```
 
-There are a few functions we can use to interrogate data in R to determine its type:
+The extra arguments after the filename help R get its hands around our data; `header=TRUE` lets it know there's a title row, and `sep=","` lets it know our columns are separated by `,` characters. Note that if you like to maintain the English idiom and put spaces after your commas, you'll need to use `sep=", "`; either is fine, as long as you're consistent.
```

Comment:
From the way I was taught, I feel like `read.csv` is a more common function. It sets the `header` and `sep` values automatically.  So that could be replaced, and much of L35 removed.
That said, perhaps part of the reason for keeping `read.table` in here is to have this discussion of its arguments? If this were a first introduction to functions, I think that would make sense. On the other hand, if the goal here is to get students working with data ASAP then best to just use `read.csv`